### PR TITLE
Gnocchi discovery mode

### PR DESCRIPTION
**Pull request by Georgehe4**

### CODE REVIEW

File: mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala [hunk 1]
```diff
@@ -37,4 +36,9 @@ object GenericFilter {
     }
   }
 
+  def regionReformat(window: Long, rdd: RDD[(ReferenceRegion, Any)]): RDD[(ReferenceRegion, Long)] = {
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala]: comment this function. maybe call chunkByRegion

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 2]
```diff
@@ -718,6 +724,30 @@ class VizReads(protected val args: VizReadsArgs) extends BDGSparkCommand[VizRead
...
+    def discoverGnocchi(): List[ReferenceRegion] = {
+
+      // filtering for features
+      val featureRegions: Option[RDD[(ReferenceRegion, Long)]] = {
+        if (!VizReads.featuresExist) {
+          log.warn("specified discovery predicate for features but no variant files were provided")
+          None
+        } else {
+          var features: RDD[Feature] = sc.parallelize[(Feature)](Array[(Feature)]())
+          VizReads.featureData.get.files.foreach(fp => features = features.union(FeatureMaterialization.load(sc, None, fp).rdd))
```
Comment (Georgehe4) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: This will look through all the feature files right now.

Comment (akmorrow13) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Ok we should fix this then. If we are going to differentiate discoverGnocci() from discoverFeatures() the methods should be distinct

File: mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala [hunk 3]
```diff
@@ -37,4 +45,16 @@ object GenericFilter {
     }
   }
 
+  /**
+   * Reformat regions to fit window size
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala]: Can you put a description of why this is necessary (i.e. this is for grouping counts to be displayed in the frontend or something)

File: mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala [hunk 4]
```diff
@@ -37,4 +45,16 @@ object GenericFilter {
...
+   *
+   * @param window binning window
+   * @param rdd Reference region rdd to reformat
+   * @return Returns a reformatted RDD with regions that fit region size
+   */
+  def chunkByRegion(window: Long, rdd: RDD[(ReferenceRegion, Any)]): RDD[(ReferenceRegion, Long)] = {
+
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/filters/GenericFilter.scala]: rm line

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala (hunk 2 above)
Comment (akmorrow13) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: test

**Pull request closed without merging**